### DISCUSSION TITLE
Is there an api to check the availability of the resources used in the storyboard file?

### DISCUSSION THREAD

**Discussion by rakuyoMo** [SwiftGen/SwiftGen#924] [General]
Just like the `validate` method provided by `R.swift`, you can check if the images and colors referenced in the storyboard are valid at runtime, or any other time.

Does `SwiftGen` have a function like this? I didn't find any documentation about it, if so please reply, thanks a lot!

**Comment by djbe** [comment]
No it does not currently have such functionality.
- SwiftGen is unaware of your project's structure, you tell it where everything is at.
- R.swift analyzes your project and tries to guess where your resources are.

**Reply by rakuyoMo** [reply]
Not just a source of files

Runtime files like storyboard, for example, only catch some errors when the controller is initialized, such as missing `@IBOutlet` linkage.

`R.swift` provides a method to initialize all storyboards, could SwiftGen add such a method to ensure the security of resources?
